File: server/src/models/VideoSource.ts
import { Model, DataTypes } from 'sequelize';
import sequelize from '../db/connection';
import PlaylistItem from './PlaylistItem';

class VideoSource extends Model {
  declare id: number;
  declare playlistItemId: number;
  declare url: string;
  declare label: string;
  declare createdTime: Date;
  declare lastActiveTime: Date;
}

VideoSource.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  playlistItemId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: PlaylistItem,
      key: 'id'
    }
  },
  url: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  createdTime: {
    type: DataTypes.DATE(3),
    allowNull: false,
    defaultValue: DataTypes.NOW
  },
  lastActiveTime: {
    type: DataTypes.DATE(3),
    allowNull: false,
    defaultValue: DataTypes.NOW
  }
}, {
  sequelize,
  modelName: 'VideoSource',
  tableName: 'video_sources',
  timestamps: false
});

export default VideoSource; 

File: server/internal/models/video_source.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// VideoSource represents a video source URL for a playlist item
type VideoSource struct {
	ID             uint           `gorm:"primaryKey;autoIncrement" json:"id"`
	PlaylistItemID uint           `gorm:"not null;index" json:"playlistItemId"`
	URL            string         `gorm:"type:varchar(255);not null" json:"url"`
	Label          string         `gorm:"type:varchar(100)" json:"label"`
	CreatedTime    time.Time      `gorm:"not null;autoCreateTime:milli" json:"createdTime"`
	LastActiveTime time.Time      `gorm:"not null;autoUpdateTime:milli" json:"lastActiveTime"`
	DeletedAt      gorm.DeletedAt `gorm:"index" json:"-"`

	// Associations
	PlaylistItem *PlaylistItem `gorm:"foreignKey:PlaylistItemID" json:"playlistItem,omitempty"`
}

// TableName specifies the table name for VideoSource model
func (VideoSource) TableName() string {
	return "video_sources"
}
